lms/djangoapps/courseware/views/views.py

```diff
@@ -444,46 +444,6 @@ def jump_to(request, course_id, location):
     return redirect(redirect_url)
 
 
-@ensure_csrf_cookie
-@ensure_valid_course_key
-@data_sharing_consent_required
-def course_root(request, course_id):
-    """
-    Renders the configured course home or redirects to the correct url. This
-    behaviour can be controlled by the following variables:
-
-    * DISABLE_UNIFIED_COURSE_TAB_FLAG WaffleFlag
-        redirects / to the unified course experience.
-    * COURSE_HOME_BEHAVIOUR SiteConfiguration:
-        can redirect from / to any course related page. Some possible redirects
-        are: 'info', 'about' and 'courseware'.
-        If this fails to find a reverse, it uses the course info page as a
-        default.
-
-    If none of those are set, this redirects to course_info (the previous
-    default behaviour).
-    """
-    course_key = CourseKey.from_string(course_id)
-
-    # If the unified course experience is enabled, redirect to the "Course" tab
-    if not DISABLE_UNIFIED_COURSE_TAB_FLAG.is_enabled(course_key):
-        return redirect(reverse(course_home_url_name(course_key), args=[course_id]))
-
-    course_home_behaviour = configuration_helpers.get_value('COURSE_HOME_BEHAVIOUR')
-    if course_home_behaviour:
-        try:
-            course_home_url = reverse(course_home_behaviour, kwargs={'course_id': course_id})
-            return redirect(course_home_url)
-        except:
-            logging.warn(
-                'COURSE_HOME_BEHAVIOUR incorrectly configured. The request url doesn\'t exist: %s',
-                course_home_behaviour
-            )
-
-    # Default behaviour if variable isn't set
-    return course_info(request, course_id)
-
-
 def course_info(request, course_id):
```

Comment:
Do we need to add the three decorators related to csrf cookie, valid course key, etc. [same as the maple.3](https://github.com/open-craft/edx-platform/blob/2212934c3620c6acb465073e246847d5dcfb9f48/lms/djangoapps/courseware/views/views.py#L447-L450) branch, since we removed the `course_root` function.

Reply:
Good catch fixed!